### PR TITLE
feat: use Pressable in TouchableRipple

### PR DESCRIPTION
**Pull request by szymonrybczak**
This PR replaces `Touchables` with `Pressable` within `TouchableRipple`.

#### Related issues: 
- #3396

### CODE REVIEW

**Review by hurali97 — CHANGES_REQUESTED** [review]
Thanks for raising this PR, much appreciated 🌟.

There are a couple changes that can be addressed.

**Comment by hurali97** [comment]
@szymonrybczak Additionally, there are some regressions that causes instability on web. I am attaching the screenshots of the relevant components that are affected by the PR changes. If you can take a look at these, it will be so helpful.

### Platform(web):

- The back icon button mis aligns on web.

<details>
 <summary>Bottom-Navigation</summary>

 - layout of bottom-navigation shrinks on web.

Bottom-Navigation
</details>

<details>
 <summary> Buttons </summary>

- Ripple doesn't respect the border-radius.

Buttons
</details>

<details>
 <summary>Icon-Buttons</summary>

- Doesn't center the icon correctly, and ripple only happens on icon, not the whole component.

Icon-Buttons
</details>

<details>
 <summary>Radio-Button-Group</summary>

- Ripple is not round, as compared to the baseline.

Radio-Button-Group
</details>

<details>
 <summary> SearchBar </summary>

- Related to Icon Buttons.

SearchBar
</details>

<details>
 <summary>Segmented-Buttons</summary>

- Related to the border-radius in Buttons.

Segmented-Buttons
</details>

<details>
 <summary>Text-Input</summary>

- Related to Icon Buttons.

Text-Input
</details>

<details>
 <summary>Toggle-Buttons</summary>

- Ripple doesn't fill the container, it only fills the inner view.

Toggle-Buttons
</details>

<details>
 <summary>Touchable-Ripple-Difference</summary>
<br />
Touchable Ripple New:

Touchable-Ripple-New

<br />
Touchable Ripple Baseline:

Touchable-Ripple-Old
</details>

File: src/components/TouchableRipple/TouchableRipple.native.tsx [hunk 1]
```diff
@@ -58,32 +58,45 @@ const TouchableRipple = ({
...
       {...rest}
       disabled={disabled}
-      style={[borderless && styles.overflowHidden, style]}
-      underlayColor={calculatedUnderlayColor}
+      style={[
+        borderless && styles.overflowHidden,
+        showUnderlay && { backgroundColor: calculatedUnderlayColor },
+        style,
+      ]}
+      onPressIn={(e: GestureResponderEvent) => {
```
[the remark this answers is not in the document]
Comment (szymonrybczak) [reply src/components/TouchableRipple/TouchableRipple.native.tsx:89]: Fixed ✅

File: src/components/TouchableRipple/TouchableRipple.native.tsx [hunk 2]
```diff
@@ -58,32 +58,45 @@ const TouchableRipple = ({
...
+        setShowUnderlay(true);
+        rest.onPressIn?.(e);
+      }}
+      onPressOut={(e: GestureResponderEvent) => {
```
[the remark this answers is not in the document]
Comment (szymonrybczak) [reply src/components/TouchableRipple/TouchableRipple.native.tsx:93]: Fixed ✅

**Comment by szymonrybczak** [comment]
@hurali97 Yes, sure. This was really the problem, I fixed it! 😅

**Comment by hurali97** [comment]
@szymonrybczak Thanks for incorporating them real quick. 🌟

@lukewalczak I'll re-run the tests and re-review the PR tomorrow morning.

**Review by hurali97 — APPROVED** [review]

**Review by lukewalczak — APPROVED** [review]
Great work 🎉 Thanks @szymonrybczak for your effort and @hurali97 for reviewing and testing the PR!

**Pull request merged**